compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -277,6 +277,11 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 let discr = self.discriminant_for_variant(op.layout.ty, variant)?;
                 self.write_immediate(*discr, &dest)?;
             }
+
+            WrapUnsafeBinder(ref op, _ty) => {
```

Comment:
Constructing an unsafe binder acts like a transmute since the operand's layout does not change.

Reply:
Please make that a comment in the code :)

Reply:
Oli had a better idea which is to just replace it with a transmute cast, I'll try that first then it's unreachable